src/main/kotlin/io/github/codeocean_scraper/PageFetcher.kt

```diff
@@ -1,12 +1,24 @@
 package io.github.codeocean_scraper
 
 import org.jsoup.Jsoup
+import java.net.URL
+
+private val cookieName = "_hands-on-programming_session"
 
 class PageFetcher(val sessionKey: String) {
     fun fetch(url: String)
         = Jsoup.connect(url)
-               .cookie("_hands-on-programming_session", sessionKey)
+               .cookie(cookieName, sessionKey)
                .maxBodySize(128 * 1024 * 1024)
                .timeout(0)
                .get()
 }
+
+fun isValidSessionKey(baseUrl: URL, sessionKey: String): Boolean {
+    // Sign in page redirects to homepage if session key is valid
+    val signInUrl = URL(baseUrl, "/sign_in")
+    val response = Jsoup.connect(signInUrl.toString())
+                        .cookie(cookieName, sessionKey)
+                        .execute()
+    return response.url() == baseUrl
```

Comment:
Does kotlin rewrite `==` to `.equals()`?

Thats of course also a way to detect redirects... :D